ql/termstructures/volatility/equityfx/andreasenhugevolatilityadapter.hpp

```diff
@@ -0,0 +1,58 @@
+/* -*- mode: c++; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- */
+
+/*
+ Copyright (C) 2017 Klaus Spanderen
+
+ This file is part of QuantLib, a free-software/open-source library
+ for financial quantitative analysts and developers - http://quantlib.org/
+
+ QuantLib is free software: you can redistribute it and/or modify it
+ under the terms of the QuantLib license.  You should have received a
+ copy of the license along with this program; if not, please email
+ <quantlib-dev@lists.sf.net>. The license is also available online at
+ <http://quantlib.org/license.shtml>.
+
+ This program is distributed in the hope that it will be useful, but WITHOUT
+ ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ FOR A PARTICULAR PURPOSE.  See the license for more details.
+*/
+
+/*! \file andreasenhugevolatilityadapter.hpp
+    \brief Implements the BlackVolTermStructure interface based on a
+           Andreasen-Huge volatility interpolation
+*/
+
+#ifndef quantlib_andreasen_huge_volatility_adapter_hpp
+#define quantlib_andreasen_huge_volatility_adapter_hpp
+
+#include <ql/termstructures/volatility/equityfx/blackvariancecurve.hpp>
+
+namespace QuantLib {
+    class AndreasenHugeVolatilityInterpl;
+
+    class AndreasenHugeVolatilityAdapter : public BlackVarianceTermStructure {
+      public:
+        AndreasenHugeVolatilityAdapter(
+            const boost::shared_ptr<AndreasenHugeVolatilityInterpl>& volInterpl,
+            Real eps = 1e-10);
```

Comment:
It turns out that for some combination of inputs, compiler, the way brew compiled boost, and possibly the phase of the moon, the failing test finds a fixed point of the iteration with an error around 1.8e-10.

Do you see any objections to increasing this to 1e-9 or 1e-8?  The other tests seem to pass either way.

Reply:
Thanks you for your help on this issue!  The current default value of 1e-10 is far too pedantic. Seems that QL is normally using 1e-6, e.g. in blackFormulaImpliedStdDev. I have adjusted the default and relaxed the tests.